README.rst

```diff
@@ -409,6 +413,26 @@ Number equality with substitutions:
     assertThis: '{numberOne}'
     assertEquals: '{numberTwo}' # substituted numbers not equal. Stop pipeline.
 
+GOTCHA:
+  A string formatting expression like ``{numberOne}`` will always return
+  a string. This means a comparison to a number type will fail even if it looks
+  like the same number.
+
+  .. code-block:: yaml
+
+    meaningOfLife: 42
+    assertThis: '{meaningOfLife}'
+    assertEquals: 42 # stop pipeline. '42' != 42 (str vs int)
+
+  vs
+
+  .. code-block:: yaml
+
+    meaningOfLife: 42
```

Comment:
ixnay to both. The whole point of the example is to demonstrate even if you put in an `int`, it turns into a `str` as a result of the formatting expression.